todo_app/app.py

```diff
@@ -1,37 +1,131 @@
-from flask import Flask, render_template, request, redirect, url_for
-import Trello_items as trello
+from flask import Flask, render_template, request, redirect, url_for, abort
+from flask_login import LoginManager, login_required
+import flask_login
+from flask_login.mixins import UserMixin
+from oauthlib.oauth2 import WebApplicationClient
+import Mongo_items as mongo
 import viewmodel as vm
+import os, requests, json
+from functools import wraps
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object('flask_config.Config')
 
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        client = WebApplicationClient(os.getenv("CLIENT_ID"))
+        client.state = client.state_generator()
+        authredirect = client.prepare_request_uri("https://github.com/login/oauth/authorize", state=client.state)
+        return redirect(authredirect)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        user = User(user_id)
+        return user
+
+    login_manager.init_app(app)
+
+    class User(UserMixin):
+        def __init__(self, userid):
+            self.id = userid
+            if self.id == "69510597" :
+                self.role = 'writer'
+            else :
+                self.role = 'reader'
+
+    def checkrole(f):
```

Comment:
checkrole is an unclear name for what this decorator does. Better would be something like `writer_required`